src/NLogShipper/LogzioTarget.cs

```diff
@@ -73,11 +77,24 @@ protected override void Write(LogEventInfo logEvent)
                     {"@timestamp", logEvent.TimeStamp.ToString("o")},
                     {"logger", logEvent.LoggerName},
                     {"level", logEvent.Level.Name},
-                    {"message", _usingDefaultLayout ? logEvent.FormattedMessage : RenderLogEvent(Layout, logEvent)},
                     {"exception", logEvent.Exception?.ToString()},
+                    {"message", _usingDefaultLayout ? logEvent.FormattedMessage : RenderLogEvent(Layout, logEvent)},
                     {"sequenceId", logEvent.SequenceID.ToString()}
                 };
-
+                if (Format.ToLower() == "json")
+                {
+                    try
+                    {
+                        foreach (var keyValuePair in JObject.Parse(values["message"].ToString()))
```

Comment:
Can you show a use-case where this is better than just adding the wanted `ContextProperty`-items to the NLog-Target ?

Reply:
You mean use something like setproperty(property, value) to the logger instead of append to values dictionary?
Does it matter?
Can you give an example of your usage?

Reply:
I was just curious of the "driver" for this change, but probably not important.